src/fmu/dataio/readers.py

```diff
@@ -0,0 +1,112 @@
+"""Module for reading data not supported by other tools"""
+
+from __future__ import annotations
+
+import hashlib
+import json
+from dataclasses import dataclass, field
+from pathlib import Path
+
+
+def read_faultroom_file(filename: str | Path) -> FaultRoomSurface:
+    """Faultroom surface data are (geo)JSON file or dicts; needs separate handling.
+
+    Faultroom data are quite propriatary data and is not supported by e.g. xtgeo
+    currently. Hence it is read locally. As input, both a dict and a file with
+    extension .json og .geojson can be applied
+    """
+    filename = Path(filename)
+    if "json" in filename.suffix.lower():
+        # try read the (geo)json file
+        with open(filename, "r", encoding="utf-8") as stream:
+            dict_obj = json.load(stream)
+
+        if (
+            "metadata" in dict_obj
+            and "source" in dict_obj["metadata"]
+            and "FaultRoom" in dict_obj["metadata"]["source"]
+        ):
+            return FaultRoomSurface(dict_obj)
+
+    raise ValueError(
+        f"Cannot read faultroom file. Check if file <{filename}> really is "
+        "on FaultRoom format, and has extension 'json' or 'geojson'"
+    )
+
+
+@dataclass
+class FaultRoomSurface:
+    """Parse the requested props from FaultRoom plugin output format."""
+
+    storage: dict
+
+    horizons: list = field(default_factory=list, init=False)
+    faults: list = field(default_factory=list, init=False)
+    juxtaposition_fw: list = field(default_factory=list, init=False)
+    juxtaposition_hw: list = field(default_factory=list, init=False)
+    properties: list = field(default_factory=list, init=False)
+    bbox: dict = field(default_factory=dict, init=False)
+    name: str = field(default="", init=False)
+    tagname: str = field(default="faultroom", init=False)
+
+    def __post_init__(self) -> None:
+        self._get_horizons()
+        self._get_faults()
+        self._get_juxtaposition()
+        self._get_properties()
+        self._get_bbox()
+        self._derive_names()
+
+    def _get_horizons(self) -> None:
+        self.horizons = self.storage["metadata"].get("horizons")
+
+    def _get_faults(self) -> None:
+        self.faults = self.storage["metadata"]["faults"].get("default")
+
+    def _get_juxtaposition(self) -> None:
+        self.juxtaposition_fw = self.storage["metadata"]["juxtaposition"].get("fw")
+        self.juxtaposition_hw = self.storage["metadata"]["juxtaposition"].get("hw")
+
+    def _get_properties(self) -> None:
+        self.properties = self.storage["metadata"].get("properties")
+
+    def _get_bbox(self) -> None:
```

Comment:
`_get` to me implies the method returns something; maybe `_set` might be more accurate`

Reply:
Good catch; changed it